pkg/node/manager/manager_test.go

```diff
@@ -854,32 +855,31 @@ func TestNodeManagerEmitStatus(t *testing.T) {
 
 		done := make(chan struct{})
 		reattempt := make(chan struct{})
-		checkStatus := func() ([]types.Status, <-chan struct{}) {
```

Comment:
Commenting on random line in manager_test.go.

I wonder if it would be simpler if that part:
```
https://github.com/cilium/cilium/blob/e4e8a15fd783fd0524752090d9fffaa76ff5fa28/pkg/node/manager/manager.go#L362-L367
```
was pulled into `singleBackgroundLoop` and:

In that case, you wouldn't need to override global `baseBackgroundSyncInterval = 1 * time.Millisecond`. 
After calling `singleBackgroundLoop` you could use `Get` from statedb instead of `GetWatch` which would simplify the test a bit, kind of similar to the current test here: https://github.com/cilium/cilium/blob/e4e8a15fd783fd0524752090d9fffaa76ff5fa28/pkg/node/manager/manager_test.go#L442

Reply:
yeah, fair point - I think I'll leave it as is for the merge to go through as it fixes a bug, but took a note to revisit this once I have a bit more cycles